internal/webhooks/clusterclass.go

```diff
@@ -123,8 +142,90 @@ func (webhook *ClusterClass) validate(old, new *clusterv1.ClusterClass) error {
 	// Ensure spec changes are compatible.
 	allErrs = append(allErrs, check.ClusterClassesAreCompatible(old, new)...)
 
+	// Ensure no MachineDeploymentClass currently in use has been removed from the ClusterClass.
+	allErrs = append(allErrs, webhook.validateRemovedMachineDeploymentClassesAreNotUsed(old, new)...)
+
 	if len(allErrs) > 0 {
 		return apierrors.NewInvalid(clusterv1.GroupVersion.WithKind("ClusterClass").GroupKind(), new.Name, allErrs)
 	}
 	return nil
 }
+
+func (webhook *ClusterClass) validateRemovedMachineDeploymentClassesAreNotUsed(old, new *clusterv1.ClusterClass) field.ErrorList {
+	var allErrs field.ErrorList
+	if old == nil {
+		return allErrs
+	}
+	removedClasses := webhook.removedMachineClasses(old, new)
+
+	// If no classes have been removed return early as no further checks are needed.
+	if len(removedClasses) == 0 {
+		return allErrs
+	}
+
+	// Retrieve all clusters using the ClusterClass.
+	clusters, errs := webhook.getClustersUsingClusterClass(old)
+	if len(errs) > 0 {
+		allErrs = append(allErrs, errs...)
+		return allErrs
+	}
+
+	// Create a set of records of machineDeploymentClass information where the name is a fully qualified name for the
+	// MachineDeployment topology in the form cluster.Name/machineDeploymentTopology.Name and links to the machineDeploymentClass Name.
+	for _, c := range clusters {
+		for _, machineDeploymentTopology := range c.Spec.Topology.Workers.MachineDeployments {
+			if removedClasses.Has(machineDeploymentTopology.Class) {
+				// TODO(killianmuldoon): Improve error printing here so large scale changes don't flood the error log e.g. deduplication, only example usages given.
+				allErrs = append(allErrs, field.Forbidden(field.NewPath(""), fmt.Sprintf("MachineDeploymentClass %v is in use in MachineDeployment %v in Cluster %v. ClusterClass %v modification not allowed",
```

Comment:
I think the path should be the path to the MD array (i.e. ~ `.spec.topology.workers.machineDeployments`)

Reply:
```suggestion
				allErrs = append(allErrs, field.Forbidden(field.NewPath(""), fmt.Sprintf("MachineDeploymentClass %v is in use in MachineDeployment topology %v in Cluster %v. ClusterClass %v modification not allowed",
```
I think we have to mention topology, otherwise it sounds like its the name of the MachineDeployment